### PR TITLE
PC-1464: Improve select your supplier question

### PR DESCRIPTION
**Pull request by samyou-softwire**
[Link to Jira ticket](https://beisdigital.atlassian.net/browse/PC-1464)

# Description

adds an alternative supplier page

if the user selects that they don't know they are sent here and can select any supplier

needs some extra work when displaying supplier name to ensure the correct supplier is always shown, as well as some additional work when submitting so energy suppliers see the alternative that was picked

# Checklist

- [x] I have made any necessary updates to the documentation
- [x] I have checked there are no unnecessary IDE warnings in this PR
- [x] I have checked there are no unintentional line ending changes
- [x] I have added tests where applicable
- [x] If I have made any changes to the code, I have used the IDE auto-formatter on it
- [x] If I have made any changes to Python files (even if not changing any content strings), I have run `make extract-translations`
- [x] If I have made any changes to website flow, I have updated the [Flow Miro Board](https://miro.com/app/board/uXjVK4AgQ3E=/)
- [x] If I have made any changes to website flow, I have checked forward and back behaviour is still consistent

# Screenshots

image
image
image
image

### CODE REVIEW

**Comment by samyou-softwire** [comment]
noticed I've missed error texts, will push a fix

File: help_to_heat/frontdoor/interface.py [hunk 1]
```diff
@@ -162,37 +169,52 @@ class SupplierConverter:
...
 
-    def _get_supplier(self):
-        return api.session.get_answer(self.session_id, "supplier")["supplier"]
+    def get_supplier(self):
+        supplier = api.session.get_answer(self.session_id, supplier_page).get(supplier_field)
+        alternative_supplier = api.session.get_answer(self.session_id, alternative_supplier_page).get(
+            alternative_supplier_field
+        )
+
+        return alternative_supplier if self._did_specify_alternative() else supplier
```
Comment (jdgage) [inline help_to_heat/frontdoor/interface.py:178]: I don't think there's real overhead to `api.session.get_answer` is there? We could be quite a bit more efficient with how many times we're calling it if so.

File: help_to_heat/frontdoor/interface.py [hunk 2]
```diff
@@ -162,37 +169,52 @@ class SupplierConverter:
...
+                alternative_supplier_field
+            )
+            session_data[supplier_field] = alternative_supplier
+
         if self._is_bulb() or self._is_shell():
-            session_data["supplier"] = "Octopus Energy"
+            session_data[supplier_field] = "Octopus Energy"
         if self._is_utility_warehouse():
-            session_data["supplier"] = "E.ON Next"
+            session_data[supplier_field] = "E.ON Next"
```
Comment (jdgage) [inline help_to_heat/frontdoor/interface.py:217]: Should these be using constants?

File: help_to_heat/frontdoor/routing/forwards_routing.py [hunk 3]
```diff
@@ -246,9 +252,8 @@ def _country_next_page(answers):
     return _unknown_response
 
 
-@_requires_answer(supplier_field)
-def _supplier_next_page(answers):
-    supplier = answers.get(supplier_field)
+# where to send the user post them picking a supplier
+def _post_supplier_pick_next_page(supplier):
```
Comment (jdgage) [inline help_to_heat/frontdoor/routing/forwards_routing.py:256]: Can this be named better so the comment isn't necessary?

File: help_to_heat/frontdoor/consts.py [hunk 4]
```diff
@@ -104,7 +108,7 @@
 supplier_field_shell = "Shell"
 supplier_field_utilita = "Utilita"
 supplier_field_utility_warehouse = "Utility Warehouse"
-supplier_fields = [
+supplier_fields_real = [
```
Comment (jdgage) [inline help_to_heat/frontdoor/consts.py:111]: Can this be made to read a bit more naturally? Would quibble that these aren't fields, they're values

**Review by jdgage — CHANGES_REQUESTED** [review]

File: help_to_heat/frontdoor/interface.py (hunk 2 above)
Comment (samyou-softwire) [reply help_to_heat/frontdoor/interface.py:217]: yep

File: help_to_heat/frontdoor/interface.py (hunk 1 above)
Comment (samyou-softwire) [reply help_to_heat/frontdoor/interface.py:178]: don't think so too much, though it is a database call. have reworked it to just check once

File: help_to_heat/frontdoor/routing/forwards_routing.py (hunk 3 above)
Comment (samyou-softwire) [reply help_to_heat/frontdoor/routing/forwards_routing.py:256]: looking back I don't think the comment adds much anyway. have removed it & changed the name slightly

File: help_to_heat/frontdoor/consts.py (hunk 4 above)
Comment (samyou-softwire) [reply help_to_heat/frontdoor/consts.py:111]: for what its worth most of these fields could be named as such, ie `supplier_field_utilitia` -> `supplier_field_value_utilita`. for now have renamed the `fields` to `field_values`

**Comment by samyou-softwire** [comment]
translations for "Select an alternative energy supplier" have been added

**Comment by samyou-softwire** [comment]
@jdgage if there's nothing else & pipelines pass this should be ready to go

**Review by jdgage — APPROVED** [review]

**Pull request merged**